### PR TITLE
[IMP] website: introduce `s_image_frame` snippet

### PR DESCRIPTION
**Pull request by xlu-odoo**
This commit adds the new `s_image_frame` snippet.

<details>
<summary>Snippet Preview</summary>

image

</details>

task-4105328
Part-of: task-4077427

requires: https://github.com/odoo/design-themes/pull/864

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/views/snippets/s_image_frame.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+<template id="s_image_frame" name="Image - Frame">
+    <section class="s_image_frame o_cc o_cc2 pt64 pb64">
+        <div class="container">
+            <div class="row o_grid_mode" data-row-count="11">
+                <div class="o_grid_item g-height-4 g-col-lg-5 col-lg-6 rounded-2 o_colored_level o_cc o_cc1" style="z-index: 1; grid-area: 7 / 7 / 11 / 12; --grid-item-padding-x: 36px; --grid-item-padding-y: 36px;">
+                    <h3>Product highlight</h3>
+                    <p class="lead">Choose a vibrant image and write an inspiring paragraph about it. It does not have to be long, but it should reinforce your image.</p>
+                </div>
+                <div class="o_grid_item o_grid_item_image d-none d-md-block g-height-12 g-col-lg-6 col-lg-6 rounded-2 overflow-hidden oe_img_bg o_colored_level o_cc o_cc1 " style="grid-area: 1 / 1 / 12 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
```
Comment (chgo-odoo) [inline addons/website/views/snippets/s_image_frame.xml:12]: `g-col-lg-*` class doesn't match the grid-area you defined, it should be `g-col-lg-12`
```suggestion
                <div class="o_grid_item o_grid_item_image d-none d-md-block g-height-12 g-col-lg-12 col-lg-6 rounded-2 overflow-hidden oe_img_bg o_colored_level o_cc o_cc1 " style="grid-area: 1 / 1 / 12 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
```

Comment (chgo-odoo) [inline addons/website/views/snippets/s_image_frame.xml:12]: `rounded-2` is not a thing as it doesn't mimic the edition behaviour, you should replicate the web editor behaviour thus `rounded`and `border-radius=""`
```suggestion
                <div class="o_grid_item o_grid_item_image d-none d-md-block g-height-12 g-col-lg-6 col-lg-6 rounded overflow-hidden oe_img_bg o_colored_level o_cc o_cc1 " style="grid-area: 1 / 1 / 12 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;  border-radius: 6.4px !important;">
```

Comment (chgo-odoo) [inline addons/website/views/snippets/s_image_frame.xml:12]: Is there a reason for adding `overflow-hidden` here ? It doesn't seem to apply anything and it is not editable by the user, you should probably remove it.

File: addons/website/views/snippets/s_image_frame.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+<template id="s_image_frame" name="Image - Frame">
+    <section class="s_image_frame o_cc o_cc2 pt64 pb64">
```
Comment (chgo-odoo) [inline addons/website/views/snippets/s_image_frame.xml:5]: Suggestion but I'd switch to `o_cc_1`. The main point is that due to the padding applied on the section, it looks buggy on mobile.

image

File: addons/website/views/snippets/s_image_frame.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        <div class="container">
+            <div class="row o_grid_mode" data-row-count="11">
+                <div class="o_grid_item g-height-4 g-col-lg-5 col-lg-6 rounded-2 o_colored_level o_cc o_cc1" style="z-index: 1; grid-area: 7 / 7 / 11 / 12; --grid-item-padding-x: 36px; --grid-item-padding-y: 36px;">
+                    <h3>Product highlight</h3>
+                    <p class="lead">Choose a vibrant image and write an inspiring paragraph about it. It does not have to be long, but it should reinforce your image.</p>
+                </div>
+                <div class="o_grid_item o_grid_item_image d-none d-md-block g-height-12 g-col-lg-6 col-lg-6 rounded-2 overflow-hidden oe_img_bg o_colored_level o_cc o_cc1 " style="grid-area: 1 / 1 / 12 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
+                    <img src="/web/image/website.s_image_frame_default_image" class="img img-fluid mx-auto" alt=""/>
```
Comment (chgo-odoo) [inline addons/website/views/snippets/s_image_frame.xml:13]: To be tested but I'd switch the image to a background one, to fix the white padding in columns layout. 

Doing that will also allow you to add padding on both element, ensuring a better visual result in columns mode.

If the aspect-ratio of the image affects the layout, ignore this comment.

image

File: addons/website/views/snippets/s_image_frame.xml (hunk 1 above)
Comment (chgo-odoo) [inline addons/website/views/snippets/s_image_frame.xml:12]: Don't hide the image on mobile, the whole purpose of the images category is to show images 😉 

```suggestion
                <div class="o_grid_item o_grid_item_image d-md-block g-height-12 g-col-lg-6 col-10 offset-1 offset-lg-0 col-lg-6 rounded-2 overflow-hidden oe_img_bg o_colored_level o_cc o_cc1 " style="grid-area: 1 / 1 / 12 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
```

**Review by chgo-odoo — CHANGES_REQUESTED** [review]
Thanks @xlu-odoo, nice job here ! 💪

I added some small comments, but I think there is room for improvement, we can discuss it at the office so it's easier !

| This PR | Suggestion |
|--------|--------|
| image | image |

File: addons/website/views/snippets/s_image_frame.xml (hunk 1 above)
Comment (xlu-odoo) [reply addons/website/views/snippets/s_image_frame.xml]: as discussed, ability to control the radius in WE (instead of just having `rounded` or not)

**Review by chgo-odoo — APPROVED** [review]
LGTM, thanks @xlu-odoo 👍

File: addons/website/views/snippets/s_image_frame.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+<template id="s_image_frame" name="Image Frame">
+    <section class="s_image_frame o_colored_level o_cc o_cc1 pt64 pb64">
+        <div class="container">
+            <div class="row o_grid_mode" data-row-count="11">
+                <div class="o_grid_item o_grid_item_image g-height-12 g-col-lg-12 col-10 col-lg-12 offset-1 offset-lg-0 order-lg-0 rounded overflow-hidden o_colored_level o_cc o_cc1" style="grid-area: 1 / 1 / 12 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px; border-radius: 6.4px !important; order: 1;">
```
Comment (SergeBayet) [inline addons/website/views/snippets/s_image_frame.xml:8]: @xlu-odoo g-height-11 no? (12-1 = 11)

**Review by SergeBayet — CHANGES_REQUESTED** [review]
LGTM :+1: Just a little miscalculation in my opinion. See below.

**Review by SergeBayet — APPROVED** [review]

**Comment by sobo-odoo** [comment]
@robodoo delegate=SergeBayet

**Comment by SergeBayet** [comment]
@robodoo r+

**Pull request closed without merging**